src/otaclient/ota_core.py

```diff
@@ -1201,14 +1206,11 @@ def client_update(self, request: ClientUpdateRequestV2) -> None:
                 client_update_control_flags=self._client_update_control_flags,
                 metrics=self._metrics,
             ).execute()
-        except ota_errors.OTAError as e:
-            self._live_ota_status = OTAStatus.FAILURE
-            self._on_failure(
-                e,
-                ota_status=OTAStatus.FAILURE,
-                failure_reason=e.get_failure_reason(),
-                failure_type=e.failure_type,
-            )
+        except ota_errors.OTAError:
+            # TODO(airkei) [2025-06-19]: should return the dedicated error code for "client update"
+            self._exit_from_dynamic_client()
+        except Exception:
+            self._exit_from_dynamic_client()
```

Comment:
Not set status in client update at the moment, just restart and back to the original otaclient.